core/GlobalState.cc

```diff
@@ -1327,6 +1327,11 @@ MethodRef GlobalState::enterMethodSymbol(Loc loc, ClassOrModuleRef owner, NameRe
     if (store.exists()) {
         ENFORCE(store.isMethod(), "existing symbol is not a method");
         counterInc("symbols.hit");
+        if (!store.locs(*this).empty() && absl::c_count(store.locs(*this), loc) == 0) {
+            ENFORCE(!symbolTableFrozen);
+            store.asMethodRef().data(*this)->addLoc(*this, loc);
```

Comment:
I'm not sure that we want to call `addLoc` here. I'm a little worried about what effect this will actually have, as it will affect literally every call to `enterMethodSymbol`, and I think that could potentially cause some non-obvious effects.

I'd prefer being more cautious and doing something at the call site, like what we did when we fixed a related bug with singleton class locs in this PR:

#6447